llarp/rpc/rpc_server.cpp

```diff
@@ -94,6 +94,210 @@ namespace llarp::rpc
     }
   }
 
+  void
+  HandleJSONCommand(
+      oxenmq::Message& msg,
+      std::string replyCmd,
+      std::function<void(nlohmann::json, ReplyFunction_t)> handleRequest)
+  {
+    const auto maybe = MaybeParseJSON(msg);
+    if (not maybe.has_value())
+    {
+      msg.send_back(replyCmd, CreateJSONError("failed to parse json"));
+      return;
+    }
+    if (not maybe->is_object())
+    {
+      msg.send_back(replyCmd, CreateJSONError("data not a json object"));
+      return;
+    }
+    try
+    {
+      handleRequest(*maybe, [defer = msg.send_later(), cmd = replyCmd](std::string result) {
+        defer(cmd, result);
+      });
```

Comment:
Could move the above cases in here and then `throw std::runtime_error{"bad stuff"}` so that all the error cases DRY the `send_back` to one place in the `catch`.

Reply:
no we can't because this lambda can be called from the logic thread or a random zmq thread depending on what branch is taken.

Reply:
(this is why the defer is there to begin with)